modules/common/src/main/java/com/opengamma/sdk/common/ServiceInvoker.java

```diff
@@ -242,4 +246,26 @@ public void close() {
     executor.shutdown();
   }
 
+  /**
+   * Constructs the user agent to be added as a header to all HTTP requests, based on the application version.
+   *
+   * @return The user agent.
+   */
+  public static String generateUserAgent() {
+    Properties systemProperties = System.getProperties();
+    return "opengamma-sdk-java/" +
+        SDK_VERSION +
+        " (" +
+        systemProperties.getProperty("os.name") +
+        " " +
+        systemProperties.getProperty("os.version") +
+        " " +
+        systemProperties.getProperty("os.arch") +
+        ")" +
```

Comment:
You don't normally see two sets of parentheses immediately next to each other. The user agent string is normally of the form:

```
product1/version1 (comment1; comment2) product2/version2 (comment3; comment4)
...
```

where products are listed in decreasing order of relevance to the user agent. 

I think we have two products here: the SDK itself, and Java. This suggests a user agent string like:

```
opengamma-sdk-java/1.1.0-SNAPSHOT (Mac OS X; 10.12.4; x86_64) Java/1.8.0_112 (Oracle Corporation)
```

although you could argue the OS details belong with Java.